dask/tests/test_sizeof.py

```diff
@@ -147,3 +149,34 @@ def test_dict():
     d = {i: x for i in range(100)}
     assert sizeof(d) > x.nbytes * 100
     assert isinstance(sizeof(d), int)
+
+
+def test_plugin_hook():
+    class _CustomType:
+        pass
+
+    def hook(obj):
+        raise NotImplementedError("This hook should not be called")
+
+    def plugin(dispatch):
+        dispatch.register(_CustomType, hook)
+
+    mod_name = "dask_sizeof"
+    plugin_name = "plugin"
+
+    # Create test module
+    mod = types.ModuleType(mod_name)
+    setattr(mod, plugin_name, plugin)
+    sys.modules[mod_name] = mod
```

Comment:
this `sys.modules` patch is permanent

to avoid leaking test state you can use a subprocess like dask/distributed :
https://github.com/dask/distributed/blob/0b10b031c0d3958d389ca45ffa5736d596093202/distributed/comm/tests/test_comms.py#L1313-L1327

or mock.patch/monkeypatch like pandas https://github.com/pandas-dev/pandas/blob/5b2fb093f6abd6f5022fe5459af8327c216c5808/pandas/tests/plotting/test_backend.py#L47-L62